Deserialize.scala

```diff
@@ -0,0 +1,167 @@
+// Author: James Dunn, UC Berkeley (dunn@eecs.berkeley.edu)
```

Comment:
Make sure to add a license file and match the license comments at the top of files (look at examples in BOOM/Gemmni/etc).